drivers/i2c/CMakeLists.txt

```diff
@@ -18,8 +18,12 @@ zephyr_library_sources_ifdef(CONFIG_I2C_SBCON		i2c_sbcon.c)
 zephyr_library_sources_ifdef(CONFIG_I2C_NIOS2		i2c_nios2.c)
 zephyr_library_sources_ifdef(CONFIG_I2C_STM32_V1        i2c_ll_stm32_v1.c)
 zephyr_library_sources_ifdef(CONFIG_I2C_STM32_V2        i2c_ll_stm32_v2.c)
-zephyr_library_sources_codegen_ifdef(CONFIG_I2C_STM32_V1 i2c_ll_stm32.c SEARCH_PATH ${PROJECT_SOURCE_DIR}"/templates/")
-zephyr_library_sources_codegen_ifdef(CONFIG_I2C_STM32_V2 i2c_ll_stm32.c SEARCH_PATH ${PROJECT_SOURCE_DIR}"/templates/")
+
+zephyr_library_sources_ifdef(CONFIG_I2C_STM32_V1        i2c_ll_stm32.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_STM32_V2        i2c_ll_stm32.c)
+
+zephyr_library_sources_jinjagen_ifdef(CONFIG_I2C_STM32_V1 i2c_ll_stm32_device.h.jinja2 SEARCH_PATH ${PROJECT_SOURCE_DIR}"/templates/")
```

Comment:
I can drop the SEARCH_PATH and make it hard coded to comply with the requirements, but I think it a good feature to driver developers point to places where they store commonly used templates.

Reply:
I think we should have the ${PROJECT_SOURCE_DIR}"/templates/" should be included by default since that will be common templates we define for driver developers.

Reply:
+1

Also, it should be ${ZEPHYR_BASE}/templates

Reply:
Ok, ${ZEPHYR_BASE}/templates is added by default, any other folder can still be added through the same mechanism.